File: 1496-path-crossing/1496-path-crossing.cpp
class Solution {
public:
    bool isPathCrossing(string path) {
        set<pair<int, int>> st;
        int row=0, col=0;
        st.insert({row, col});
        int n = path.size();

        for(int i=0; i<n; i++){
            if(path[i] == 'N') row++;
            else if(path[i] == 'E') col++;
            else if(path[i] == 'W') col--;
            else if(path[i] == 'S') row--;

            if(st.find({row, col}) != st.end()) return true;
            st.insert({row, col});
        }

        return false;
    }
};

File: 1496-path-crossing/1496-path-crossing.java
class Solution {
    public boolean isPathCrossing(String path) {
        Set<String> set = new HashSet();
        int row=0, col=0;
        set.add(row+","+col);

        for(char c: path.toCharArray()){
            if(c == 'N') row++;
            else if(c == 'E') col++;
            else if(c == 'W') col--;
            else if(c == 'S') row--;

            String p = row+","+col;
            if(set.contains(p)) return true;
            set.add(p);
        }
        return false;
    }
}